portal-api/src/v1/users/routes.js

```diff
@@ -26,7 +26,27 @@ const userService = new UserService();
 const signInLinkService = new SignInLinkService(randomGenerator, hasher, userRepository, userService);
 const signInLinkController = new SignInLinkController(signInLinkService);
 
-module.exports.list = (req, res, next) => {
+const isAdmin = (req, res, next) => {
+  if (req.user && req.user.roles.includes('admin')) {
+    // User is an admin, allow access to the route
+    next();
+  } else {
+    // User is not an admin, send a 403 Forbidden response
+    res.status(403).json({ error: 'Unauthorized' });
+  }
+};
+
+const isUserOrAdmin = (req, res, next) => {
```

Comment:
As mentioned in a call -- I think it might be nicer if we kept everything in our routes: `.\portal-api\src\v1\routes.js` and use our existing middleware `validateUserHasAtLeastOneAllowedRole({ allowedRoles: [ADMIN] })` for when we need to check that a user is admin.

We'll need to write some new middleware following the same pattern as `validateUserHasAtLeastOneAllowedRole` to add in validation for `validateUserIsAdminOrIsActingOnSelf`. For now, we could try ``validateUserIsActingOnSelfToChangePasswordOrIsAdmin`. Ideally, we could later abstract that to be `validateUserIsAdminOrIsActingOnSelf({actingOnSelfRules: changingPasswordRule})` which would be great and reusable.

We should look to add tests for the middleware `validateUserIsAdminOrIsActingOnSelf` in much the same way as `validateUserHasAtLeastOneAllowedRole` already has.

We could also look at adding `withValidateUserHasAtLeastOneAllowedRole` tests to our api tests, much like we have `withValidateUsernameAndEmailMatchTests`, but I think that might be outside the scope of this ticket for now